### ISSUE TITLE
Denial-of-service lockup in txg_sync when several raw recvs happen at once

### ISSUE THREAD

**Issue by numinit** [openzfs/zfs#13040]
### System information

Type | Version/Name
 --- | ---
Distribution Name	| NixOS / Ubuntu
Distribution Version	| 21.11 / 20.04.3
Kernel Version	| 5.10.90 / 5.11.0
Architecture	| x86_64
OpenZFS Version	| 2.1.2 / 2.1.1

### Describe the problem you're observing

As indicated in [this thread](https://www.reddit.com/r/zfs/comments/rkuf7i/pool_continues_to_become_suspended_000_errors_on/), another user and I happen to have a similar disk configuration (raidz1 with 3x WD Red CMR and 1x Seagate; raidz2 with 4x WD Red CMR and 2x Seagate) and are getting the same lockup when performing multiple raw recvs in parallel. This only started appearing in NixOS 21.11 when using the Syncoid module and when using znapzend, since the module authors switched to performing sends in parallel. Therefore, a system performing a send can lock up a remote system accepting recvs:

```
# zpool status -v backuptank
  pool: backuptank
 state: SUSPENDED
status: One or more devices are faulted in response to IO failures.
action: Make sure the affected devices are connected, then run 'zpool clear'.
   see: https://openzfs.github.io/openzfs-docs/msg/ZFS-8000-HC
  scan: resilvered 10.2G in 00:17:58 with 0 errors on Sat Dec 11 22:49:49 2021
config:

        NAME                                     STATE     READ WRITE CKSUM
        backuptank                               ONLINE       0     0     0
          raidz2-0                               ONLINE       0     0     0
            scsi-SATA_WDC_WD100EFAX-68_<serial>  ONLINE       0     0     0
            scsi-SATA_WDC_WD100EFAX-68_<serial>  ONLINE       0     0     0
            scsi-SATA_WDC_WD100EFAX-68_<serial>  ONLINE       0     0     0
            scsi-SATA_ST10000VN0004-1Z_<serial>  ONLINE       0     0     0
            scsi-SATA_WDC_WD101EFAX-68_<serial>  ONLINE       0     0     0
            scsi-SATA_ST10000VN0004-1Z_<serial>  ONLINE       0     0     0
```

This is reproducible 100% of the time on my machine. With enough sends done in parallel, txg_sync threads get blocked and the pool eventually suspends. The only way out is a `reboot -f` of the entire system. If I had to guess, this seems like a timing dependent race condition that is likely due to similar disk timings on both of our machines.

The workaround is simple, just do one raw recv at a time.

### Describe how to reproduce the problem

Do several ZFS raw recvs from a remote pool in parallel. Eventually, you may get a lockup in txg_sync.

### Include any warning/errors/backtraces from the system logs

```
[303649.029094] INFO: task txg_sync:1595 blocked for more than 120 seconds.
[303649.029136]       Tainted: P           OE     5.11.0-43-generic #47~20.04.2-Ubuntu
[303649.029165] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[303649.029195] task:txg_sync        state:D stack:    0 pid: 1595 ppid:     2 flags:0x00004000
[303649.029200] Call Trace:
[303649.029206]  __schedule+0x44c/0x8a0
[303649.029217]  ? zio_taskq_member.isra.0.constprop.0+0x60/0x60 [zfs]
[303649.029339]  schedule+0x4f/0xc0
[303649.029453]  schedule_timeout+0x18f/0x290
[303649.029457]  ? __next_timer_interrupt+0x130/0x130
[303649.029462]  io_schedule_timeout+0x1e/0x50
[303649.029466]  __cv_timedwait_common+0x134/0x170 [spl]
[303649.029474]  ? wait_woken+0x80/0x80
[303649.029478]  __cv_timedwait_io+0x19/0x20 [spl]
[303649.029486]  zio_wait+0x137/0x280 [zfs]
[303649.029590]  dsl_pool_sync+0xce/0x4f0 [zfs]
[303649.029690]  spa_sync+0x562/0xff0 [zfs]
[303649.029797]  ? mutex_lock+0x13/0x40
[303649.029801]  ? spa_txg_history_init_io+0x106/0x110 [zfs]
[303649.029910]  txg_sync_thread+0x2c6/0x460 [zfs]
[303649.030013]  ? txg_thread_exit.isra.0+0x60/0x60 [zfs]
[303649.030110]  thread_generic_wrapper+0x79/0x90 [spl]
[303649.030120]  ? __thread_exit+0x20/0x20 [spl]
[303649.030129]  kthread+0x12b/0x150
[303649.030133]  ? set_kthread_struct+0x40/0x40
[303649.030136]  ret_from_fork+0x22/0x30
[303649.030191] INFO: task zfs:3459005 blocked for more than 120 seconds.
[303649.030216]       Tainted: P           OE     5.11.0-43-generic #47~20.04.2-Ubuntu
[303649.030244] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[303649.030271] task:zfs             state:D stack:    0 pid:3459005 ppid:3458509 flags:0x00004000
```

**Comment by numinit** [comment]
Possibly related to #9130. I submitted a new issue because they seem to be getting random lockups while we've traced it down to near-100% reproducibility with lots of parallel raw recvs.

**Comment by numinit** [comment]
Is it possible that a couple disks that are slower (or faster) than the rest may be aggravating this issue?
